rir/src/compiler/pir/type.cpp

```diff
@@ -75,6 +80,13 @@ void PirType::merge(SEXPTYPE sexptype) {
     case WEAKREFSXP:
     case S4SXP:
         t_.r = val().t_.r;
+        t_.r.reset(RType::missing);
+        t_.r.reset(RType::unbound);
```

Comment:
this makes no sense at all.

If you look at the constructor `PirType(SEXP)`, then it does a number of additional things already, including that.